src/ens/resolvers/index.ts

```diff
@@ -0,0 +1,57 @@
+import { GraphQLScalarType, Kind } from 'graphql';
+import * as _ from 'lodash';
+import { EthqlAccount, EthqlBlock, EthqlTransaction } from '../../core/model';
+import blockResolvers from '../../core/resolvers/block';
+import rootResolvers from '../../core/resolvers/root';
+
+import Web3 = require('web3');
+import { EthqlContext } from '../../context';
+
+function isEnsDomain(input: string) {
+  return input.includes('.eth');
+}
+
+// tslint:disable-next-line
+const Address = new GraphQLScalarType({
+  name: 'Address',
+  description: 'An account address or ENS domain',
+  serialize: String,
+  parseValue: input => (Web3.utils.isAddress(input) || isEnsDomain(input) ? input : undefined),
+  parseLiteral: ast => {
+    if (ast.kind !== Kind.STRING || (!Web3.utils.isAddress(ast.value) && !isEnsDomain(ast.value))) {
```

Comment:
We actually want to return a `Promise<String>` in all cases, whether it is an ENS domain or not. If it's an ETH address, we'd return `Promise.resolve(ast.value)`, whereas if it's an ENS domain, we'd return the promise from the ENS lib.

Then the wrapper resolvers (see above) would resolve the promise before delegating onto the original resolver.

Reply:
@raulk I'm not sure of how to implement it this way, as context/services are not accessible.